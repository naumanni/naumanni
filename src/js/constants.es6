// Mastodon's max message size
export const MASTODON_MAX_CONTENT_SIZE = 500

export const NAUMANNI_VERSION = `${process.env.NAUMANNI_VERSION || 'development'}`

export const REGEX_PGP_FINGERPRINT = /PGP Key Fingerprint: ([0-9a-fA-F]+)/

export const COLUMN_TIMELINE = 'timeline'
export const COLUMN_FRIENDS = 'friends'
export const COLUMN_TALK = 'talk'
export const COLUMN_NOTIFICATIONS = 'notifications'
export const COLUMN_TAG = 'hashtag'

export const TIMELINE_FEDERATION = 'federation'
export const TIMELINE_LOCAL = 'local'
export const TIMELINE_HOME = 'home'

export const TIMELINE_FILTER_BOOST = 'timeline_filter_boost'
export const TIMELINE_FILTER_REPLY = 'timeline_filter_reply'

export const DIALOG_ADD_ACCOUNT = 'addAccount'
export const DIALOG_AUTHORIZE_ACCOUNT = 'authorizeAccount'
export const DIALOG_MEDIA_VIEWER = 'mediaViewer'
export const DIALOG_GLOBAL_PREFERENCES = 'preferences'
export const DIALOG_USER_DETAIL = 'userDetail'
export const DIALOG_WELCOME = 'welcome'

export const SUBJECT_MIXED = '!!__SUBJECT_MIXED__!!'   // Symbolだとjsonizeできないのでダメ

export const VISIBLITY_DIRECT = 'direct'
export const VISIBLITY_PRIVATE = 'private'
export const VISIBLITY_UNLISTED = 'unlisted'
export const VISIBLITY_PUBLIC = 'public'

export const NOTIFICATION_TYPE_MENTION = 'mention'
export const NOTIFICATION_TYPE_REBLOG = 'reblog'
export const NOTIFICATION_TYPE_FAVOURITE = 'favourite'
export const NOTIFICATION_TYPE_FOLLOW = 'follow'

export const KEY_TAB = 9
export const KEY_ENTER = 13  // 0x0D
export const KEY_ESC = 27    // 0x1B
export const KEY_ARROW_UP = 38
export const KEY_ARROW_DOWN = 40

export const ACCT_PATTERN = '[a-zA-Z0-9_]+@[a-zA-Z0-9\.\-]+[a-zA-Z0-9]+'
export const ACCT_REX = new RegExp(ACCT_PATTERN)
export const MESSAGE_TAG_REX = /--NEM\.([0-9a-f]{8})\.(\d+)\/(\d+)--/

export const STREAM_HOME = 'user'
export const STREAM_LOCAL = 'public:local'
export const STREAM_FEDERATION = 'public'
export const STREAM_TAG = 'hashtag'

// MastodonのWebsocketから送られてくるframeのevent types
export const EVENT_UPDATE = 'update'
export const EVENT_NOTIFICATION = 'notification'

// WebsocketManagerのlistnerに送られるevent type
export const WEBSOCKET_EVENT_ERROR = 'error'
export const WEBSOCKET_EVENT_OPEN = 'open'
export const WEBSOCKET_EVENT_MESSAGE = 'message'
export const WEBSOCKET_EVENT_CLOSE = 'close'

// statusをparseしたあとのToken
export const TOKEN_TEXT = 'text'
export const TOKEN_BREAK = 'break'
export const TOKEN_URL = 'url'
export const TOKEN_MENTION = 'mention'
export const TOKEN_HASHTAG = 'hashtag'
export const TOKEN_EMOJI = 'emoji'

// 最下部からこの位置に来たら、AutoPagingを開始する
export const AUTO_PAGING_MARGIN = 300

// Timelineで普段保持するStatus数
export const MAX_STATUSES = 40

// localStorageのKey
export const STORAGE_KEY_LAST_SEND_FROM = 'naumanni::last_send_from'
export const STORAGE_KEY_PREFERENCES = 'naumanni::preferences'
export const STORAGE_KEY_TOOT_VISIBILITY = 'naumanni::toot_visibility'

export const LOCALES = Object.freeze({
  en: 'English',
  ja: '日本語',
})


// naumanniおすすめのInstance
export const NAUMANNI_PREFERRED_INSTANCES = [
  ['mastodon.social'],
  ['mastodon.cloud'],
  ['mastodon.xyz'],
  ['pawoo.net', 'https://pawoo.net/favicon.png'],
  ['mstdn.jp'],
  ['friends.nico'],
]
