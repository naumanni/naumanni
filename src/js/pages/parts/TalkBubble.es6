/* @flow */
import React from 'react'
import {List} from 'immutable'
import {FormattedDate} from 'react-intl'

import {IconFont, SafeContent} from './'


type Props = {
  createdAt: Date,
  isEncrypted: boolean,
  parsedContent: List<{[type: string]: {[string]: any}}>,
  onClickHashTag: (string, SyntheticEvent) => void,
}

export default class TalkBubble extends React.PureComponent {
  props: Props

  render() {
    const {createdAt, isEncrypted, parsedContent, onClickHashTag} = this.props

    return (
      <li>
        <div className={`status-content ${isEncrypted ? 'is-encrypted' : ''}`}>
          <SafeContent parsedContent={parsedContent} onClickHashTag={onClickHashTag} />
        </div>
        <div className="status-date">
          <FormattedDate value={createdAt}
            year="numeric" month="2-digit" day="2-digit"
            hour="2-digit" minute="2-digit" second="2-digit"
          />
        </div>
        {isEncrypted && <div className="status-isEncrypted"><IconFont iconName="lock" /></div>}
      </li>
    )
  }
}
