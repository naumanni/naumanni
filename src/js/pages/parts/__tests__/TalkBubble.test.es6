import React from 'react'
import {mountWithIntl} from 'enzyme-react-intl'
import serializer from 'enzyme-to-json/serializer'
import {List} from 'immutable'

import {TOKEN_HASHTAG} from 'src/constants'
import TalkBubble from '../TalkBubble'


expect.addSnapshotSerializer(serializer)

describe('TalkBubble', () => {
  let defaultProps = {
    createdAt: new Date(2017, 7, 28),
    isEncrypted: false,
    parsedContent: List([
      {
        type: TOKEN_HASHTAG,
        tag: 'naumanni',
        url: 'https://friends.nico/tags/naumanni',
      }
    ]),
    onClickHashTag: jest.fn(),
  }

  describe('snapshot', () => {
    it('isEncrypted == false', () => {
      const wrapper = mountWithIntl(<TalkBubble {...defaultProps} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('isEncrypted == true', () => {
      const props = {
        ...defaultProps,
        isEncrypted: true,
      }
      const wrapper = mountWithIntl(<TalkBubble {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  it('propagate hash tag link click event', () => {
    const onClickHashTag = jest.fn()
    const props = {
      ...defaultProps,
      onClickHashTag,
    }
    const wrapper = mountWithIntl(<TalkBubble {...props} />)

    const hashtag = wrapper.find('a')
    hashtag.simulate('click')

    expect(onClickHashTag).toHaveBeenCalled()
  })
})
