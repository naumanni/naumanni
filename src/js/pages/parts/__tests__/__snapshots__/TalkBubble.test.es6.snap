// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalkBubble snapshot isEncrypted == false 1`] = `
<TalkBubble
  createdAt={2017-08-27T15:00:00.000Z}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  isEncrypted={false}
  onClickHashTag={[Function]}
  parsedContent={
    Array [
      Object {
        "tag": "naumanni",
        "type": "hashtag",
        "url": "https://friends.nico/tags/naumanni",
      },
    ]
  }
>
  <li>
    <div
      className="status-content "
    >
      <SafeContent
        onClickHashTag={[Function]}
        parsedContent={
          Array [
            Object {
              "tag": "naumanni",
              "type": "hashtag",
              "url": "https://friends.nico/tags/naumanni",
            },
          ]
        }
      >
        <p>
          <a
            onClick={[Function]}
          >
            #
            naumanni
          </a>
        </p>
      </SafeContent>
    </div>
    <div
      className="status-date"
    >
      <FormattedDate
        day="2-digit"
        hour="2-digit"
        minute="2-digit"
        month="2-digit"
        second="2-digit"
        value={2017-08-27T15:00:00.000Z}
        year="numeric"
      >
        <span>
          08/28/2017, 12:00:00 AM
        </span>
      </FormattedDate>
    </div>
  </li>
</TalkBubble>
`;

exports[`TalkBubble snapshot isEncrypted == true 1`] = `
<TalkBubble
  createdAt={2017-08-27T15:00:00.000Z}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  isEncrypted={true}
  onClickHashTag={[Function]}
  parsedContent={
    Array [
      Object {
        "tag": "naumanni",
        "type": "hashtag",
        "url": "https://friends.nico/tags/naumanni",
      },
    ]
  }
>
  <li>
    <div
      className="status-content is-encrypted"
    >
      <SafeContent
        onClickHashTag={[Function]}
        parsedContent={
          Array [
            Object {
              "tag": "naumanni",
              "type": "hashtag",
              "url": "https://friends.nico/tags/naumanni",
            },
          ]
        }
      >
        <p>
          <a
            onClick={[Function]}
          >
            #
            naumanni
          </a>
        </p>
      </SafeContent>
    </div>
    <div
      className="status-date"
    >
      <FormattedDate
        day="2-digit"
        hour="2-digit"
        minute="2-digit"
        month="2-digit"
        second="2-digit"
        value={2017-08-27T15:00:00.000Z}
        year="numeric"
      >
        <span>
          08/28/2017, 12:00:00 AM
        </span>
      </FormattedDate>
    </div>
    <div
      className="status-isEncrypted"
    >
      <IconFont
        iconName="lock"
      >
        <span
          className="icon icon-lock"
        />
      </IconFont>
    </div>
  </li>
</TalkBubble>
`;
